pkg/gateway/inmemorywallet.go

```diff
@@ -8,35 +8,34 @@ package gateway
 
 import "errors"
 
-// InMemoryWallet stores identity information used to connect to a Hyperledger Fabric network.
+// InMemoryWalletStore stores identity information used to connect to a Hyperledger Fabric network.
 // Instances are created using NewInMemoryWallet()
-type InMemoryWallet struct {
-	idhandler IDHandler
-	storage   map[string]map[string]string
+type InMemoryWalletStore struct {
+	storage map[string][]byte
 }
 
 // NewInMemoryWallet creates an instance of a wallet, backed by files on the filesystem
-func NewInMemoryWallet() *InMemoryWallet {
-	return &InMemoryWallet{newX509IdentityHandler(), make(map[string]map[string]string, 10)}
+func NewInMemoryWallet() Wallet {
```

Comment:
Why return an interface instead of struct pointer from a constructor? I think the previous version was more correct.

Reply:
fixed